app/models/assignment.rb

```diff
@@ -25,11 +25,13 @@ class Assignment < ApplicationRecord
   validates :title, presence: true
   validates :title, length: { maximum: 60 }
   validates :title, uniqueness: { scope: :organization_id }
+  validates :title, exclusion: { in: %w[new edit],
+                                 message: 'should not be a reserved word' }
 
   validates :slug, uniqueness: { scope: :organization_id }
   validates :slug, presence: true
   validates :slug, length: { maximum: 60 }
-  validates :slug, format: { with: /\A[-a-zA-Z0-9_]+\z/,
+  validates :slug, format: { with: /\A[-a-zA-Z0-9_]*\z/,
```

Comment:
The reason I did this is sort of buried above. With the old regex, the user would be displayed this validation failure AND the presence validation failure if a slug wasn't present. Now we only tell them the format is incorrect if there's actually a slug to check.